File: lib/env/request.rb
class Katamari::Env::Request
  import org.jboss.netty.util.CharsetUtil
  
  attr_accessor :headers, :uri, :body, :method, :path, :params

  def initialize(request)
    @_request, @headers, @params, @uri, @method, @body = request, {}, {}, request.uri, request.method.name.downcase.to_sym, request.content.to_string(0, request.content.readable_bytes, CharsetUtil::UTF_8)
    @_request.headers.each { |h| @headers[h.key] = h.value }
  end

  def origin
    @_request
  end
end

File: src/main/java/io/katamari/env/Request.java
package io.katamari.env;

import java.util.Map;
import java.util.HashMap;

import io.netty.handler.codec.http.DefaultFullHttpRequest;
import io.netty.handler.codec.http.HttpMethod;
import io.netty.handler.codec.http.HttpVersion;
import io.netty.buffer.ByteBuf;

public class Request extends DefaultFullHttpRequest {
  private final Map<String,String> params = new HashMap<String,String>();
  private String path;

  public Request(HttpVersion httpVersion, HttpMethod method, String uri) {
    super(httpVersion, method, uri);
  }

  public Request(HttpVersion httpVersion, HttpMethod method, String uri, ByteBuf content) {
    super(httpVersion, method, uri, content);
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public Map<String,String> getParams() {
    return params;
  }

  public Object getParam(String key) {
    return params.get(key);
  }

  public void setParam(String key, String value) {
    params.put(key, value);
  }
}